src/include/serial_inline.h

```diff
@@ -300,32 +299,13 @@ __wt_update_serial(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_PAGE *page
         return (0);
 
     /*
-     * We would like to call __wt_txn_update_oldest only in the event that there are further updates
-     * to this page, the check against WT_TXN_NONE is used as an indicator of there being further
-     * updates on this page.
+     * Look for obsolete updates if:
+     * 1. The page is configured to evict soon.
```

Comment:
I don't understand this. If the page would be evict soon, why we spend time to check the obsolete updates on it? It should be the opposite right?

Reply:
It is possible that the page is marked for eviction soon, but it is possible to evict this page if it is being accessed concurrently by many application threads. In those situations, checking for obsolete updates can be helpful.

Reply:
Can you add a comment? This is counter intuitive to me. Is there any data backing that this is useful?

Reply:
I added a comment to explain the need for page eviction soon check.